compiler/rustc_mir/src/interpret/step.rs

```diff
@@ -266,10 +266,7 @@ impl<'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
                 let layout = self.layout_of(ty)?;
                 if layout.is_unsized() {
                     // FIXME: This should be a span_bug (#80742)
-                    self.tcx.sess.delay_span_bug(
-                        self.frame().current_span(),
-                        &format!("SizeOf nullary MIR operator called for unsized type {}", ty),
-                    );
+                    throw_inval!(SizeOfUnsizedType(ty));
```

Comment:
Please keep the `delay_span_bug`, and then throw the error after that. (That's what we do for `transmute`.)

Reply:
Done.